pkg/engine/engine.go

```diff
@@ -300,10 +334,25 @@ func (e *Engine) initialize(ctx context.Context, options ...Option) error {
 	if err != nil {
 		return fmt.Errorf("failed to initialize LRU cache: %w", err)
 	}
+	const (
+		// detectableChunksChanMultiplier is set to accommodate a high number of concurrent worker goroutines.
+		// This multiplier ensures that the detectableChunksChan channel has sufficient buffer capacity
+		// to hold messages from multiple worker groups (detector workers/ verificationOverlap workers) without blocking.
+		// A large buffer helps accommodate for the fact workers are producing data at a faster rate
+		// than it can be consumed.
```

Comment:
You raise a good point. If the producers are generating data faster than the consumers can process it, there is a risk of the consumers becoming a bottleneck, which could limit overall throughput. This would be true regardless of buffer size.

Simply increasing the buffer capacity provides only a temporary solution, as the buffer will eventually fill up if the consumers can't keep pace. A more robust approach may be to scale the number of consumer goroutines to match the production rate. This helps ensure the consumers have enough total capacity to handle all of the data being produced in parallel. This would likely require a bit more testing and fine-tuning so I took the faster approach in the interim.